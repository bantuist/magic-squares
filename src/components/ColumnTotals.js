import React from 'react';
import styled from 'styled-components';

const Cell = styled.div`
  height: 100%;
  width: 100px;
  display: flex;
  justify-content: center;
  align-items: center;
`;
const Columns = styled.div`
  font-size: 1.5em;
  grid-column: 1/4;
  display: flex;
  flex-direction: row;
`;
const Total = styled.span`
  color: ${({ isCorrectTotal }) => isCorrectTotal ? '#7AE582' : '#FF5A5F'};
`;

const ColumnTotals = ({ columns, diagonals, onIsCorrectTotal }) => {
  const columnTotals = Object.keys(columns).map(key => {
    const total = columns[key];

    return (
      <Cell key={key} className="total">
        <Total isCorrectTotal={onIsCorrectTotal(total)}>{total ? total : '' }</Total>
      </Cell>
    );
  });

  const d1Total = diagonals[0];
  const d2Total = diagonals[1];

  return (
    <Columns>
      <Cell>
        <Total isCorrectTotal={onIsCorrectTotal(diagonals[0])}>{d1Total ? d1Total : '' }</Total>
      </Cell>
      {columnTotals}
      <Cell>
        <Total isCorrectTotal={onIsCorrectTotal(diagonals[1])}>{d2Total ? d2Total : '' }</Total>
      </Cell>
    </Columns>
  );
};

export default ColumnTotals;