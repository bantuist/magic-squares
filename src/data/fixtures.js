export const activeElement = 'grid-cell cell-0';
export const cells = {
  "0": {
    "diagonal": "1",
    "id": 0,
    "value": 0,
    "x": 3,
    "y": 1
  },
  "1": {
    "diagonal": "non-diagonal",
    "id": 1,
    "value": 0,
    "x": 3,
    "y": 2
  },
  "2": {
    "diagonal": "2",
    "id": 2,
    "value": 0,
    "x": 3,
    "y": 3
  },
  "3": {
    "diagonal": "non-diagonal",
    "id": 3,
    "value": 0,
    "x": 2,
    "y": 1
  },
  "4": {
    "diagonal": "1",
    "id": 4,
    "value": 0,
    "x": 2,
    "y": 2
  },
  "5": {
    "diagonal": "non-diagonal",
    "id": 5,
    "value": 0,
    "x": 2,
    "y": 3
  },
  "6": {
    "diagonal": "2",
    "id": 6,
    "value": 0,
    "x": 1,
    "y": 1
  },
  "7": {
    "diagonal": "non-diagonal",
    "id": 7,
    "value": 0,
    "x": 1,
    "y": 2
  },
  "8": {
    "diagonal": "1",
    "id": 8,
    "value": 0,
    "x": 1,
    "y": 3
  }
};
export const gridSize = 3;
export const solution = [4, 9, 2, 3, 5, 7, 8, 1, 6];
export const total = 15;
export const totals = {
  "columns": {
    "1": 0,
    "2": 0,
    "3": 0
  },
  "diagonals": [0, 0],
  "rows": {
    "1": 0,
    "2": 0,
    "3": 0
  }
};