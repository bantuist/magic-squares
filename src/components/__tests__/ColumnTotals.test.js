import React from 'react';
import { shallow } from 'enzyme';
import ColumnTotals from '../ColumnTotals';
import { totals } from '../../data/fixtures';

const onIsCorrectTotal = jest.fn();
const props = {
  columns: totals.columns,
  diagonals: totals.diagonals,
  onIsCorrectTotal
};

describe('ColumnTotals', () => {
  const columnTotals = shallow(<ColumnTotals { ...props } />);

  it('renders without crashing', () => {
    expect(columnTotals).toMatchSnapshot();
  });
});