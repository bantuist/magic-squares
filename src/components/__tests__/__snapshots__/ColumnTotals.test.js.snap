// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnTotals renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ColumnTotals
    columns={
        Object {
            "1": 0,
            "2": 0,
            "3": 0,
          }
    }
    diagonals={
        Array [
            0,
            0,
          ]
    }
    onIsCorrectTotal={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <styled.span
                    isCorrectTotal={undefined}
          />
</styled.div>,
        Array [
          <styled.div
            className="total"
>
            <styled.span
                        isCorrectTotal={undefined}
            />
</styled.div>,
          <styled.div
            className="total"
>
            <styled.span
                        isCorrectTotal={undefined}
            />
</styled.div>,
          <styled.div
            className="total"
>
            <styled.span
                        isCorrectTotal={undefined}
            />
</styled.div>,
        ],
        <styled.div>
          <styled.span
                    isCorrectTotal={undefined}
          />
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <styled.span
            isCorrectTotal={undefined}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "isCorrectTotal": undefined,
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": <styled.span
            isCorrectTotal={undefined}
/>,
          "className": "total",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "isCorrectTotal": undefined,
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "children": <styled.span
            isCorrectTotal={undefined}
/>,
          "className": "total",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "isCorrectTotal": undefined,
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "children": <styled.span
            isCorrectTotal={undefined}
/>,
          "className": "total",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "isCorrectTotal": undefined,
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <styled.span
            isCorrectTotal={undefined}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "isCorrectTotal": undefined,
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <styled.span
                        isCorrectTotal={undefined}
            />
</styled.div>,
          Array [
            <styled.div
              className="total"
>
              <styled.span
                            isCorrectTotal={undefined}
              />
</styled.div>,
            <styled.div
              className="total"
>
              <styled.span
                            isCorrectTotal={undefined}
              />
</styled.div>,
            <styled.div
              className="total"
>
              <styled.span
                            isCorrectTotal={undefined}
              />
</styled.div>,
          ],
          <styled.div>
            <styled.span
                        isCorrectTotal={undefined}
            />
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <styled.span
              isCorrectTotal={undefined}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "",
              "isCorrectTotal": undefined,
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": <styled.span
              isCorrectTotal={undefined}
/>,
            "className": "total",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "",
              "isCorrectTotal": undefined,
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "children": <styled.span
              isCorrectTotal={undefined}
/>,
            "className": "total",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "",
              "isCorrectTotal": undefined,
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "children": <styled.span
              isCorrectTotal={undefined}
/>,
            "className": "total",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "",
              "isCorrectTotal": undefined,
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <styled.span
              isCorrectTotal={undefined}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "",
              "isCorrectTotal": undefined,
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
